SparkPost/2web2ui
# FE-1019 - Address UX feedback for Hibana navigation

[FE-1019](https://jira.int.messagesystems.com/browse/FE-1019)

### What Changed
- Adds new `FullBanner` component - along with supporting tests and documentation
- Adds `GlobalBanner` component that handles positioning for the rendered `FullBanner`
- Adds `GlobalBanner` Redux container component and selector to handle business logic
- Updates logic in `AccountPopover` to render the customer ID by default, email as backup
- Updates design for 'Upgrade' secondary label in `AccountPopover`
- Updates `AccountPopover` meta data color to brand orange from purple
- Adds icons to `AccountPopover` component - *some of them will need to be updated later as they have not yet been introduced in to Matchbox*
- Addresses some alignment feedback in the navigation
- Renames Hibana `Navigation` to `Header` (more accurate component name!)
- Adds new `ButtonLink` component and supporting documentation - semantically a `<button>` but looks like an `<a>` - use this instead of `javascript:void(0);`!
- Adds console warning to `PageLink` component when an `onClick` prop is passed
- Adds supporting Cypress/unit tests

### How To Test

1. Log in to a relevant account with a free plan (PM for a username or sign up for a new account)
1. Verify the 'upgrade' banner renders
1. Verify the 'Upgrade' secondary label renders in the account popover
1. Navigate to `/account/cancel` - verify on `/account/settings` and `/account/billing` the pending cancellation banner
1. Navigate to any other page and verify the 'upgrade' banner renders
1. Log in to a paid account
1. Very the upgrade banner does *not* render
1. Verify the `Upgrade` secondary label on the account popover does *not* render
1. Make sure automated testing captures relevant changes

**Note: The old pending cancellation banner for the OG theme survives - though it was technically possible to update this design, the amount of effort required seemed to exceed the value such a change would provide. If you think I'm wrong let's chat about it.** 

### To Do
- [x] Add additional test for mutual exclusivity of cancellation vs. upgrade banners
- [X] Waiting on some new icons from UX - _moving forward for now so this PR doesn't get stale_
- [x] Waiting on meeting on 5/19/20 to chat about some of the UX feedback received, particularly regarding expanding the secondary section of the navigation on hover - **UX agreed we should proceed with the current design, re: showing and hiding the secondary navigation**
